test_new_session_new_group fails on OpenIndiana

The following test consistently fails on OpenIndiana:
```
FAIL: test_new_session_new_group (tests.test.FunctionalTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "$(BUILDDIR)/tests/test.py", line 2776, in test_new_session_new_group
    self.assertTrue(event.is_set())
AssertionError: False is not true
```
~~It is easily possible that the problem is neither in `sh`, nor its tests, but in OpenIndiana instead.  Unfortunately, I do not know how to prove that.  IOW, I do not understand what exactly is the test trying to do.~~

--- a/sh.py
+++ b/sh.py
@@ -2037,7 +2037,7 @@ class OProc(object):
                 if new_session:
                     os.setsid()
                 elif new_group:
-                    os.setpgrp()
+                    os.setpgid(0, 0)
 
                 sid = os.getsid(0)
                 pgid = os.getpgid(0)

